ui_testing/testcases/basic_tests/test06_orders.py

```diff
@@ -285,49 +279,59 @@ def test007_order_search(self, small_letters):
     def test008_duplicate_main_order(self):
         """
         New: Orders with test units: Duplicate an order with test unit 1 copy
-
         LIMS-3270
         :return:
         """
-        # get the random main order data
-        main_order = self.order_page.get_random_main_order_with_sub_orders_data()
-        # select the order
-        self.order_page.click_check_box(main_order['row_element'])
+        # get random order
+        self.base_selenium.LOGGER.info('select random record')
+        orders, payload = self.orders_api.get_all_orders(limit=20)
+        data_before_duplicate_main_order = random.choice(orders['orders'])
+
+        order_no = data_before_duplicate_main_order['orderNo']
+        self.order_page.apply_filter_scenario(filter_element='orders:filter_order_no', filter_text=order_no,
+                                              field_type='text')
+
+        row = self.order_page.get_last_order_row()
+        self.orders_page.open_edit_page(row)
+        data_before_duplicate = self.order_page.get_suborder_data()
```

Comment:
you can use get_suborder() in orders api

Reply:
@0xIslamTaha regarding this comment I used
**self.orders_api.get_suborder_by_order_id(id=data_before_duplicate_main_order['id'])** 
but I have a problem here that this method return the data same as  ( status = 1, 'orders': [ 'orderNo': '84793-2020'..... &first I don't want to return the status so I need .json & second  so  I want to assert with this order number this one not working after_duplicate_order['OrderNo'] so how can I handle this

Reply:
I have a question here I tried to use get_suborder_by_order_id () but I have no idea how I can get the id ( from where I mean ) so Do you have any idea how we can handle that

Reply:
@ghada-37

Reply:
you created new order, function create order return api and payload : https://github.com/modeso-open-source/1lims-automation/blob/9314f89291e3551e3d410475f8eff3855665b281/api_testing/apis/orders_api.py#L109

Id found in api